go/vt/vttablet/endtoend/transaction_test.go

```diff
@@ -366,6 +366,110 @@ func TestShutdownGracePeriod(t *testing.T) {
 	client.Rollback()
 }
 
+func TestShutdownGracePeriodWithStreamExecute(t *testing.T) {
+	client := framework.NewClient()
+
+	err := client.Begin(false)
+	require.NoError(t, err)
+	go func() {
+		_, err = client.StreamExecute("select sleep(10) from dual", nil)
+		assert.Error(t, err)
+	}()
+
+	started := false
+	for i := 0; i < 10; i++ {
+		queries := framework.LiveQueryz()
+		if len(queries) == 1 {
+			started = true
+			break
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
+	assert.True(t, started)
+
+	start := time.Now()
+	err = client.SetServingType(topodatapb.TabletType_REPLICA)
+	require.NoError(t, err)
+	assert.True(t, time.Since(start) < 5*time.Second, time.Since(start))
+	client.Rollback()
+
+	client = framework.NewClientWithTabletType(topodatapb.TabletType_REPLICA)
+	err = client.Begin(false)
+	require.NoError(t, err)
```

Comment:
nit: may be you can continue using the old client.

Reply:
spoke offline and we decided to keep it as it is